frontend/src/models.ts

```diff
@@ -428,9 +428,9 @@ export type Time<T> = {
 export type TimeNumeric = Time<number>;
 export type TimeFormatted = Time<string>;
 
-export type SelectOption = {
+export type SelectOption<T = number> = {
   label: string;
-  value: string | number;
+  value: T;
```

Comment:
FYI: I refactored the SelectOption to process a given type instead of switching between string and number. That solved a lot of conversion issues.